[xcode11] mmptest failures

https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/10186/Test_20Report/


* Xamarin.MMP.Tests.MMPTests.Unified_HelloWorld_ShouldHaveNoWarnings(True): Run failed unexpectedly: /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/xharness/tmp-test-dir/mmptest/bin/Debug/tmp-test-dir/Xamarin.MMP.Tests.MMPTests.RunMMPTest104/bin/Release/XM45Example.app/Contents/MacOS/XM45Example Output: objc[18594]: Class AMSupportURLConnectionDelegate is implemented in both /System/Library/PrivateFrameworks/EmbeddedOSInstall.framework/Versions/A/EmbeddedOSInstall (0x7fffa47fac58) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x106102fd8). One of the two will be used. Which one is undefined.
objc[18594]: Class AMSupportURLSession is implemented in both /System/Library/PrivateFrameworks/OSPersonalization.framework/Versions/A/OSPersonalization (0x7fffa60240d8) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x106103028). One of the two will be used. Which one is undefined.
objc[18594]: Class RPStream is implemented in both /System/Library/PrivateFrameworks/EmbeddedOSInstall.framework/Versions/A/EmbeddedOSInstall (0x7fffa47fbd38) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x1061030c8). One of the two will be used. Which one is undefined.
2019-06-05 21:03:57.160 XM45Example[18594:124373447] 
Unhandled Exception:
System.DllNotFoundException: mono-native
at (wrapper managed-to-native) Interop+Sys.Stat(byte&,Interop/Sys/FileStatus&)
at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.Directory.Exists (System.String path) [0x0001e] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x000c1] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int)
at System.IO.File.Create (System.String path, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.File.Create (System.String path) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at TestCase.MainClass.Main (System.String[] args) [0x00005] in <ab72487dba0f470ea210c65aaa96fb36>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: mono-native
at (wrapper managed-to-native) Interop+Sys.Stat(byte&,Interop/Sys/FileStatus&)
at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.Directory.Exists (System.String path) [0x0001e] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x000c1] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int)
at System.IO.File.Create (System.String path, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.File.Create (System.String path) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at TestCase.MainClass.Main (System.String[] args) [0x00005] in <ab72487dba0f470ea210c65aaa96fb36>:0 

* Xamarin.MMP.Tests.MMPTests.UnifiedDebugBuilds_ShouldLinkToPartialStatic_UnlessDisabled: Run failed unexpectedly: /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/tests/xharness/tmp-test-dir/mmptest/bin/Debug/tmp-test-dir/Xamarin.MMP.Tests.MMPTests.RunMMPTest114/bin/Release/XM45Example.app/Contents/MacOS/XM45Example Output: objc[19089]: Class AMSupportURLConnectionDelegate is implemented in both /System/Library/PrivateFrameworks/EmbeddedOSInstall.framework/Versions/A/EmbeddedOSInstall (0x7fffa47fac58) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x10f0c4fd8). One of the two will be used. Which one is undefined.
objc[19089]: Class AMSupportURLSession is implemented in both /System/Library/PrivateFrameworks/OSPersonalization.framework/Versions/A/OSPersonalization (0x7fffa60240d8) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x10f0c5028). One of the two will be used. Which one is undefined.
objc[19089]: Class RPStream is implemented in both /System/Library/PrivateFrameworks/EmbeddedOSInstall.framework/Versions/A/EmbeddedOSInstall (0x7fffa47fbd38) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x10f0c50c8). One of the two will be used. Which one is undefined.
2019-06-05 21:05:55.737 XM45Example[19089:124376000] 
Unhandled Exception:
System.DllNotFoundException: mono-native
at (wrapper managed-to-native) Interop+Sys.Stat(byte&,Interop/Sys/FileStatus&)
at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.Directory.Exists (System.String path) [0x0001e] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x000c1] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int)
at System.IO.File.Create (System.String path, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.File.Create (System.String path) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at TestCase.MainClass.Main (System.String[] args) [0x00005] in <4054154749ef4350bed733d7d423bd6c>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: mono-native
at (wrapper managed-to-native) Interop+Sys.Stat(byte&,Interop/Sys/FileStatus&)
at Interop+Sys.Stat (System.ReadOnlySpan`1[T] path, Interop+Sys+FileStatus& output) [0x00028] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.FileExists (System.ReadOnlySpan`1[T] fullPath, System.Int32 fileType, Interop+ErrorInfo& errorInfo) [0x00007] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath, Interop+ErrorInfo& errorInfo) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileSystem.DirectoryExists (System.ReadOnlySpan`1[T] fullPath) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.Directory.Exists (System.String path) [0x0001e] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x000c1] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int)
at System.IO.File.Create (System.String path, System.Int32 bufferSize) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at System.IO.File.Create (System.String path) [0x00000] in <2005fb5d7e52451fa5c513f23424f43d>:0 
at TestCase.MainClass.Main (System.String[] args) [0x00005] in <4054154749ef4350bed733d7d423bd6c>:0 

* Xamarin.Tests.ProductTests.MinOSVersion(iOS,MiniPhoneOS,IOSSimulator,False): Failures
Expected: <empty>
But was: < "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libmono-profiler-log.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libmono-profiler-log.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libmonosgen-2.0.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libmonosgen-2.0.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libxamarin.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libxamarin.dylib ().", "Unexpected minOS version (expected 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libmono-native-compat.dylib ().", "Unexpected minOS version (expected 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libmono-native-compat.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libxamarin-debug.dylib ().", "Unexpected minOS version (expected 6.0.0, alternatively 6.0.0, found 8.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphonesimulator.sdk/usr/lib/libxamarin-debug.dylib ()." >

* Xamarin.Tests.ProductTests.MinOSVersion(iOS,MiniPhoneOS,IOS,True): Failures
Expected: <empty>
But was: < "Unexpected minOS version (expected 6.0.0, found 7.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphoneos.sdk/usr/lib/libmono-native-compat.dylib ().", "Unexpected minOS version (expected 6.0.0, found 7.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphoneos.sdk/usr/lib/libmono-native-compat.dylib ().", "Unexpected minOS version (expected 6.0.0, found 7.0.0) in /Users/builder/jenkins/workspace/xamarin-macios-pr-builder/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/SDKs/MonoTouch.iphoneos.sdk/usr/lib/libmono-native-compat.dylib ()." >

--- a/tests/mmptest/src/MMPTest.cs
+++ b/tests/mmptest/src/MMPTest.cs
@@ -124,26 +124,25 @@ namespace Xamarin.MMP.Tests
 			});
 		}
 
-		[TestCase (false)]
-		[TestCase (true)]
-		public void Unified_HelloWorld_ShouldHaveNoWarnings (bool release)
+		[TestCase (false, false)]
+		[TestCase (false, true)]
+		[TestCase (true, false)]
+		[TestCase (true, true)]
+		public void Unified_HelloWorld_ShouldHaveNoWarnings (bool release, bool full)
 		{
 			RunMMPTest (tmpDir => {
 				TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir);
 				test.Release = release;
+				test.XM45 = full;
 
 				// Due to https://bugzilla.xamarin.com/show_bug.cgi?id=48311 we can get warnings related to the registrar
 				Func<string, bool> hasLegitWarning = results =>
 					results.Split (Environment.NewLine.ToCharArray ()).Any (x => x.Contains ("warning") && !x.Contains ("deviceBrowserView:selectionDidChange:"));
 
-				// Mobile
-				string buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
-				Assert.IsTrue (!hasLegitWarning (buildResults), "Unified_HelloWorld_ShouldHaveNoWarnings - Mobile had warning: \n" + buildResults);
+				string buildResults;
 
-				// XM45
-				test.XM45 = true;
 				buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
-				Assert.IsTrue (!hasLegitWarning (buildResults), "Unified_HelloWorld_ShouldHaveNoWarnings - XM45 had warning: \n" + buildResults);
+				Assert.IsTrue (!hasLegitWarning (buildResults), "Unified_HelloWorld_ShouldHaveNoWarnings had warning: \n" + buildResults);
 			});
 		}
 
@@ -409,35 +408,56 @@ namespace Xamarin.MMP.Tests
 			});
 		}
 
-		[Test]
-		public void UnifiedDebugBuilds_ShouldLinkToPartialStatic_UnlessDisabled ()
+		[TestCase (false)]
+		[TestCase (true)]
+		public void UnifiedDebugBuilds_ShouldLinkToPartialStatic_UnlessDisabled_Release (bool xm45)
 		{
-			RunMMPTest (tmpDir =>
-			{
-				foreach (bool xm45 in new bool[] { false, true })
-				{
-					TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) { XM45 = xm45 };
-
-					test.Release = true;
-					string buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
-					Assert.IsFalse (buildResults.Contains ("Xamarin.Mac.registrar"), "Release build should not use partial static registrar\n" + buildResults);
-
-					test.Release = false;
-					test.CSProjConfig = "<DebugSymbols>true</DebugSymbols>";
-					buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
-					Assert.IsTrue (buildResults.Contains ("Xamarin.Mac.registrar"), "Debug build should use partial static registrar\n" + buildResults );
-
-					test.CSProjConfig = "<DebugSymbols>true</DebugSymbols><MonoBundlingExtraArgs>--registrar=dynamic</MonoBundlingExtraArgs><XamMacArch>x86_64</XamMacArch>";
-					buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
-					Assert.IsFalse (buildResults.Contains ("Xamarin.Mac.registrar"), "registrar=dynamic build should not use partial static registrar\n" + buildResults);
-
-					test.CSProjConfig = "<DebugSymbols>true</DebugSymbols><MonoBundlingExtraArgs>--registrar=partial</MonoBundlingExtraArgs><XamMacArch>x86_64</XamMacArch>";
-					buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
-					Assert.IsTrue (buildResults.Contains ("Xamarin.Mac.registrar"), "registrar=partial build should use partial static registrar\n" + buildResults);
-				}
+			RunMMPTest (tmpDir => {
+				TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) { XM45 = xm45 };
+				test.Release = true;
+				string buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
+				Assert.IsFalse (buildResults.Contains ("Xamarin.Mac.registrar"), "Release build should not use partial static registrar\n" + buildResults);
 			});
 		}
 
+		[TestCase (false)]
+		[TestCase (true)]
+		public void UnifiedDebugBuilds_ShouldLinkToPartialStatic_UnlessDisabled_Debug (bool xm45)
+		{
+			RunMMPTest (tmpDir => {
+				TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) { XM45 = xm45 };
+				test.Release = false;
+				test.CSProjConfig = "<DebugSymbols>true</DebugSymbols>";
+				var buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
+				Assert.IsTrue (buildResults.Contains ("Xamarin.Mac.registrar"), "Debug build should use partial static registrar\n" + buildResults);
+			});
+		}
+
+		[TestCase (false)]
+		[TestCase (true)]
+		public void UnifiedDebugBuilds_ShouldLinkToPartialStatic_UnlessDisabled_Debug_dynamic (bool xm45)
+		{
+			RunMMPTest (tmpDir => {
+				TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) { XM45 = xm45 };
+				test.Release = false;
+				test.CSProjConfig = "<DebugSymbols>true</DebugSymbols><MonoBundlingExtraArgs>--registrar=dynamic</MonoBundlingExtraArgs><XamMacArch>x86_64</XamMacArch>";
+				var buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
+				Assert.IsFalse (buildResults.Contains ("Xamarin.Mac.registrar"), "registrar=dynamic build should not use partial static registrar\n" + buildResults);
+			});
+		}
+
+		[TestCase (false)]
+		[TestCase (true)]
+		public void UnifiedDebugBuilds_ShouldLinkToPartialStatic_UnlessDisabled_Debug_partial (bool xm45)
+		{
+			RunMMPTest (tmpDir => {
+				TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) { XM45 = xm45 };
+				test.Release = false;
+				test.CSProjConfig = "<DebugSymbols>true</DebugSymbols><MonoBundlingExtraArgs>--registrar=partial</MonoBundlingExtraArgs><XamMacArch>x86_64</XamMacArch>";
+				var buildResults = TI.TestUnifiedExecutable (test).BuildOutput;
+				Assert.IsTrue (buildResults.Contains ("Xamarin.Mac.registrar"), "registrar=partial build should use partial static registrar\n" + buildResults);
+			});
+		}
 		//https://testrail.xamarin.com/index.php?/cases/view/234141&group_by=cases:section_id&group_order=asc&group_id=51097
 		[Test]
 		public void Unified45Build_CompileToNativeOutput ()

